### PR TITLE
[s] Golems target Free Golem ship when jaunting

### PR DESCRIPTION
**Pull request by coiax**
Not all of my jaunter changes were medical, please be more careful next time @TechnoAlchemisto and @KorPhaeron. Golem jaunter access was removed for a reason, restoring it came specifically at the cost of making them target it.

Since this is a quasi-urgent PR, I haven't done any of the other stuff I'd like to do, because restraint.

### CODE REVIEW

File: code/__HELPERS/mobs.dm [hunk 1]
```diff
@@ -319,9 +319,27 @@ Proc for attack log creation, because really why not
...
-/proc/isabductor(A)
+/proc/is_species(A, species_datum)
 	. = FALSE
 	if(ishuman(A))
 		var/mob/living/carbon/human/H = A
-		if(H.dna && istype(H.dna.species, /datum/species/abductor))
+		if(H.dna && istype(H.dna.species, species_datum))
 			. = TRUE
+
+/proc/isabductor(A)
```
Comment (Iamgoofball) [inline code/__HELPERS/mobs.dm]: :-1: no, proc call overhead is shit and should just be called with is_species

Comment (optimumtact) [inline code/__HELPERS/mobs.dm]: goof these procs are not called on hot paths, I wouldn't worry too much about it

they could be macro defines however which would eliminate the extra proc call

Comment (coiax) [inline code/__HELPERS/mobs.dm]: Can you nest macrodefines and have them multiline?

Comment (Iamgoofball) [inline code/__HELPERS/mobs.dm]: you can make it 1 line

Comment (optimumtact) [inline code/__HELPERS/mobs.dm]: @coiax I don't think so, but for the time being non of the procs here are using any of that advanced functionality, so you get the speed of a single proc call with the flexibility of later converting it from a macro to a proc seamlessly

**Pull request merged**